ooniprobe/Utility/UIDevice-Hardware.m

```diff
@@ -0,0 +1,31 @@
+#include <sys/sysctl.h>
+#import "UIDevice-Hardware.h"
+
+@interface UIDevice (Hardward)
+
+- (NSString *)modelNameForModelIdentifier:(NSString *)modelIdentifier;
+
+@end
+
+@implementation UIDevice (Hardware)
+
+- (NSString *)getSysInfoByName:(char *)typeSpecifier
```

Comment:
Since this method is only ever called with `typeSpecifier` set to `hw.machine` maybe we can just place all this code inside of the `modelIdentifier` method and explicit the `hw.machine` string to avoid an extra un-necessary layer of indirection.

Reply:
Can we avoid bikeshed too much for such a simple function? We are talking about sending an email to support 😅 I just want to merge this and move on with more important tasks

Reply:
> I just want to merge this and move on with more important tasks

I disagree with this view. While I understand why you want to do more challenging stuff, and I really sympathise with this feeling, it's also important that we try to minimise the amount of unnecessary code that we commit, in the interest of maintainability.

Reply:
As the previous function became really long I would still keep this one, so we can reuse it to get more insight from the hardware (maybe in the future) by passing different params.

Reply:
> As the previous function became really long

I would not call it really long. But I sympathize with the feeling that short functions are good.

> so we can reuse it to get more insight from the hardware

Yeah, this is also a good argument. I'm going to resolve this convo.